core/src/price.ts

```diff
@@ -110,14 +113,18 @@ export const calculateTransactionGrossProfitDate = function (auction: Auction, c
         return undefined;
     }
 
-    let steps = 0;
-    let currentValue = new BigNumber(auction.approximateUnitPrice);
+    const stepNumber = Math.ceil(
+        Math.log(auction.marketUnitPrice.dividedBy(auction.approximateUnitPrice).toNumber()) /
+            Math.log(auction.priceDropRatio.toNumber())
+    );
 
-    while (currentValue.isGreaterThan(auction.marketUnitPrice)) {
-        steps += 1;
-        currentValue = currentValue.multipliedBy(auction.priceDropRatio);
-    }
     const secondsSinceLastPriceDrop = auction.secondsBetweenPriceDrops - auction.secondsTillNextPriceDrop;
-    const secondsTillProfitable = auction.secondsBetweenPriceDrops * steps - secondsSinceLastPriceDrop;
+    const secondsTillProfitable = auction.secondsBetweenPriceDrops * stepNumber - secondsSinceLastPriceDrop;
     return addSeconds(currentDate, secondsTillProfitable);
 };
+
+export const calculateTransactionGrossProfitDate = memoizee(_calculateTransactionGrossProfitDate, {
+    maxAge: TRANSACTION_GROSS_PROFIT_DATE_CACHE_MS,
+    promise: true,
+    length: 2,
+});
```

Comment:
Since this function receives auction object which comes from the store, the auction object can potentially be:
- different object although the values are the same (especially since BigNumber values are objects themselves, recreated with `new` during the enrichment)
- or the same between the calls, but with different values

To avoid this problem of comparing objects by their reference, you need to serialise it:
https://github.com/medikoo/memoizee#cache-id-resolution-normalization

Then there will be no need for the `maxAge` since as soon as the values haven't changed, the profit time will be the same as well